### DISCUSSION TITLE
Is the wasmxfhead template currently working ?

### DISCUSSION THREAD

**Discussion by jeromelaban** [unoplatform/uno#10247] [Q&A]
_Following https://github.com/unoplatform/uno/issues/3285#issuecomment-1290597351_

---
Hi, is the wasmxfhead template currently working?

I have gone through instructions at both https://platform.uno/docs/articles/get-started-dotnet-new.html?tabs=windows#uno-platform-blank-application-uwp and https://github.com/unoplatform/Uno.Xamarin.Forms.Platform
Trying on existing Xamarin projects (which is ultimately what I am looking for) and on new empty Xamarin projects.

I have tried the template 3.1.6 (from the steps in github) and also with template 4.5.16 (the latest).

When I create the project (from dev cmd), I get 
```
The template "Uno Platform WebAssembly Head for Xamarin.Forms" was created successfully.

Processing post-creation actions...
Adding
    project(s): C:\Users\me\source\repos\XamLovesUnoOrSomething\XamLovesUnoOrSomething.Wasm
    to solution file: C:\Users\me\source\repos\XamLovesUnoOrSomething\XamLovesUnoOrSomething.sln
    solution folder:
Failed to add project(s) to the solution: File `C:\Users\me\source\repos\XamLovesUnoOrSomething\XamLovesUnoOrSomething.Wasm` not found.
Post action failed.


An update for template package 'Uno.ProjectTemplates.Dotnet::3.1.6' is available.
To update the package use:
   dotnet new install Uno.ProjectTemplates.Dotnet::4.5.16
```

It creates the project, in its folder, so I right click and add it to the solution manually, but then it does not build. The error is
```
Version conflict detected for Xamarin.Forms. Install/reference Xamarin.Forms 5.0.0.2196 directly to project XamLovesUnoOrSomething.Wasm to resolve this issue.
```

I open NuGet for the solution, find the Xamarin.Forms that is installed, and add it to my WASM project, this error goes away, but it still does not build.
I close VS, delete all "obj" and "bin" folders, reopen VS, but it still does not build, with three errors.
```
Error reading response
Generation failed: System.MissingMethodException: Method not found: 'System.ReadOnlySpan`1<Char> Microsoft.IO.Path.GetFileName(System.ReadOnlySpan`1<Char>)'.
Generation failed, error code Rejected	
```

Is there anything else I can do, or try, or is this project type just not a thing anymore?
---

**Comment by jeromelaban** [comment]
Hi @Thorocaine, the Xamarin.Forms support was an experiment, not heavily used by the community and we decided to not pursue developing for it.

What's your use case?

**Reply by jon-peel** [reply]
I have an app, still small, which was started in Xamarin.
I am thinking for a few reasons I might prefer to have it in Blazor (Web Assembly).
I thought this might work as a temporary transition.

**Reply by jon-peel** [reply]
Instead of Blazor, I am thinking of trying Uno for it's Wasm.
It will give a few benefits, mainly my View Model project should just continue to work as is.

My current distribution is UWP and Android, on Uno I could keep Android, and replace UWP with WASM.
